### ISSUE TITLE
L'ordre des questions est changé si la question est répondu ou non

### ISSUE THREAD

**Issue by wiinxt** [betagouv/mon-entreprise#2021]
Si on répond à une question, qu'on clique sur précédent, puis sur suivant alors on sera plus sur la question qu'on avait répondu.

https://user-images.githubusercontent.com/6691954/154672917-7bb877de-261a-4c80-83b7-e9ec8668bc3a.mp4

**Comment by johangirod** [comment]
Effectivement, c'est un problème dû à l'impossibilité de savoir la différence entre une règle rendue non applicable ou une règle dont la valeur dans la situation est `non`, mais qui est applicable.

Par exemple, si l'on répond "non" à la question "connaissait vous votre taux AT/MAP", puis "non" à "l'activité est-elle sans aucun risque", que l'on revient en arrière et que l'on répond "oui" pour le taux, il n'y a aucun moyen de savoir si la question sur le risque est "non" parce que c'est dans la situation ou parce qu'elle est non applicable. 

On préfère donc ne pas afficher les questions déjà répondues. 

Cela va changer avec [la prochaine version de publicodes](https://github.com/betagouv/publicodes/pull/159) : on pourra différencier une règle non applicable (null) d'une règle qui vaut "non".

**Comment by johangirod** [comment]
Après discussion avec @wiinxt, nous en avons conclu qu'il faudrait tout de même améliorer l'actuel. On pourrait sauvegarder la pile des questions passées via le bouton précédent et la restaurer lors du clique sur "suivant", tout en filtrant les questions devenues non applicables (via `UNSAFE_isNotApplicable` pour le moment).
